### PR TITLE
Model flow and config files

### PR DESCRIPTION
**Pull request by mayalinetsky**
Updates:
- Changed implementation of prepare_submission_csv to include to year-month-day--housr-minute-second
- Changed baseline implementation to match the generic flow (got a better score)

Additions:
- Added a skeleton module for each section in the flow (see model_flow.png)
- Added a main file that runs the baseline permutation from the config. You can change the config in order to see results with other paramters

### CODE REVIEW

File: evaluation.py
Comment (erantr1) [inline evaluation.py:1]: sklearn has a ready-made rmse log - https://scikit-learn.org/stable/modules/model_evaluation.html#mean-squared-log-error
Note that it's computed a bit differently.

Comment (erantr1) [inline evaluation.py:1]: sklearn has a ready-made rmse log - https://scikit-learn.org/stable/modules/model_evaluation.html#mean-squared-log-error
Note that it's computed a bit differently.

File: feature_extraction.py
Comment (erantr1) [inline feature_extraction.py:1]: Didn't understand this one

File: evaluation.py
Comment (mayalinetsky) [reply evaluation.py:1]: The function here is based on what is stated in the competition page:
https://en.wikipedia.org/wiki/Root-mean-square_deviation
image

I'll ask Lior if he meant
```
from sklearn.metrics import mean_squared_log_error
mean_squared_log_error(y_true, y_pred)
```
Or
```
from sklearn.metrics import mean_squared_log_error
mean_squared_log_error(y_true-1, y_pred-1)
```

File: feature_extraction.py
Comment (mayalinetsky) [reply feature_extraction.py:1]: See the diagram I added in the new commit.
The main thing separating between 'preprocessing' and 'feature extraction' is the input and order in the flow.

Feature extraction is ANY thing you want to apply on the raw data (the dataframe containing all columns and the target column).

Preprocessing is ANY thing you want to apply on the 'X features', see 'Unprocessed Xs i' cylinder.

**Review by yairbh — APPROVED** [review]

**Review by yairbh — APPROVED** [review]

**Pull request merged**